docs/stories/EPIC-Restore deleted messages/TF-1719-User-can-restore-deleted-messages.md

```diff
@@ -0,0 +1,58 @@
+# Summary
+
+* [Related EPIC](#related-epic)
+* [Definition](#definition)
+* [Screenshots](#screenshots)
+* [Misc](#misc)
+
+## Related EPIC
+https://github.com/linagora/tmail-flutter/issues/1719
+
+## Definition
+
+- Given that I am a Tmail user and I have logged in successfully
+- When I click on three-dot button of folder Trash, I can see a new option in context menu: Recover deleted messages 
+- I select this option, a popup will be opened:
+- In the popup, I can input fields:
+   - Deletion date: 
+      + Default option is "Last 1 year"
+      + I can select other options: Last 7 days, Last 30 days, Last 6 months, Custom range 
+      + When I select Custom range, a Date picker will be displayed that allow me to select From date and To date.
+      + The date range is limited between today to one year ago. 
+   - Reception date: 
+      + Default option is "All time"
+      + I can select other options: Last 7 days, Last 30 days, Last 6 months, Last year, Custom range 
+      + When I select Custom range, a Date picker will be displayed that allow me to select From date and To date.
+      + The date range is not limited 
+   - Subject: a text field 
+   - Recipients: I can input name/email address, the system will display a suggestion list of matched emails. I can select from the list or input the recipients mannually 
+   - Senders:  I can input name/email address, the system will display a suggestion list of matched emails. I can select from the list or input the senders mannually
+   - Has attachment: a Checkbox. If it is checked, the system will find and restore email which contains attachments
+- After input conditions, I click button Restore, the system will search for matched messages in in deleted vault. 
+- When the search is in progress, there will be a toast message :"Recovery is in progress and might take time. Please wait a few minutes." 
+- When the search is completed, the system will restore 5 (configurable by system) first messages in the result to a new system folder:"Restored messages"
```

Comment:
```suggestion
- When the recovery is completed, the system will restore 5 (configurable by system) first messages in the result to a new system folder:"Restored messages"
```

Reply:
Why only 5 first messages?

Reply:
Avoid that one can restore all messages previously deleted.

That's a way to ensure that, if your account get's hacked, the hacker can't easily read all of your deleted emails...

If that is not powerful enough the user can still contact an admin.

Reply:
we have any rate limit for the number of recovery request?

Reply:
> we have any rate limit for the number of recovery request?

Ideally yes but not in a MVP for this feature